### PR TITLE
Fix Factory reconfig/upgrade

### LINKED ISSUE #348: cvmfsexec.cfg assumed blank and recreated at every factory reconfig
**Describe the bug**
Every time the reconfigure command is executed in the factory several messages regarding the recreation of the cvmfsexec.cfg file appears.

**To Reproduce**
Run reconfigure with some entries and no cvmfs configuration.

**Expected behavior**
Those files should just be created once if necessary, or no messages should appear.

**Screenshots**
```
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_Negishi/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_Negishi/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_Negishi_Op/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_Negishi_Op/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_Rice/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_Rice/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_hansen/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_Purdue_hansen/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw2/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw2/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw4/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw4/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw4_gpu/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw4_gpu/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw6/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw6/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw6_Comet/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw6_Comet/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw6_fake/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw6_fake/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw7/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw7/cvmfsexec.cfg'
Assuming blank, and re-creating...
Error opening /var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw7_gpu/cvmfsexec.cfg: [Errno 2] No such file or directory: '/var/lib/gwms-factory/work-dir/entry_CMSHTPC_T2_US_UCSD_gw7_gpu/cvmfsexec.cfg'
```

**Info (please complete the following information):**
-   GlideinWMS version: 3.10.3
-   Priority: medium
-   Stakeholders: FactoryOps
-   Components: factory

### PR DESCRIPTION
**Pull request by mambelli**
Factory reconfig or upgrade is giving errors about missing files upgrading to 3.10.4.
The failure to reconfig was misleading (due to local changes in the test instance).

The cvmfsexec building configuration is only in the main section, not in the entries.
The dictionaries (dictfiles) have been fixed accordingly.

The warning about possible outdated builds is printed only if there are builds

During testing was discovered that `distros` was not initialized in case of exception

This fixes #348

### CODE REVIEW

File: creation/lib/cgWParamDict.py [hunk 1]
```diff
@@ -340,9 +358,11 @@ def populate(self, other=None):
...
-                            distro_info[3] + "_" + distro_info[4]
-                        )  # added protection with try-except and continue if fail
+                        if len(distro_info) == 4:  # for machine types like ppc64le, aarch64
+                            distro_arch_temp = distro_info[2] + "_" + distro_info[3]
+                        else:
+                            distro_arch_temp = distro_info[3] + "_" + distro_info[4]
+                            # added protection with try-except and continue if fail
                     except:
                         print(f"Wrong subdirectory cvmfsexec/tarballs/{distro_info}")
                         continue
```
Comment (mambelli) [inline creation/lib/cgWParamDict.py:368]: and next line
```
distro_info = distros[cvmfsexec_idx].split(".")[0].split("_")
distro_arch = "_".join(distro_info[2:])
```

Comment (mambelli) [reply creation/lib/cgWParamDict.py:368]:
```
distro_info = distros[cvmfsexec_idx].split(".")[0].split("_", 3)
platform = "-".join(distro_info[1:])
```

File: creation/lib/cgWParamDict.py [hunk 2]
```diff
@@ -369,34 +389,29 @@ def populate(self, other=None):
...
-        # put user files in stage
-        for file in self.conf.get_child_list("files"):
-            add_file_unparsed(file, self.dicts, True)
-
-        # put user attributes into config files
-        for attr in self.conf.get_child_list("attrs"):
-            # ignore attributes that need expansion in the global section
-            if str(attr.get_val()).find("$") == -1 or not self.enable_expansion:  # does not need to be expanded
-                add_attr_unparsed(attr, self.dicts, "main")
+                print("...No sources specified. Building/Rebuilding of cvmfsexec distributions disabled!")
```
Comment (mambelli) [inline creation/lib/cgWParamDict.py:394]: No need to be under if

File: creation/lib/cgWParamDict.py [hunk 3]
```diff
@@ -369,34 +389,29 @@ def populate(self, other=None):
...
+            try:
+                # fetch the on-demand cvmfs provisioning feature setting
+                ondemand_cvmfs = self.dicts["attrs"]["GLIDEIN_USE_CVMFSEXEC"]
+            except KeyError as e:
+                # on-demand CVMFS not used at the global level; ignore and continue
+                pass
+            else:
+                # check if on demand cvmfs provisioning is requested/enabled
+                if ondemand_cvmfs:
```
Comment (mambelli) [inline creation/lib/cgWParamDict.py:403]:
```
            ondemand_cvmfs = 0
            try:
                # fetch the on-demand cvmfs provisioning feature setting
                ondemand_cvmfs = self.dicts["attrs"]["GLIDEIN_USE_CVMFSEXEC"]
            except KeyError as e:
                # on-demand CVMFS not used at the global level; ignore and continue
                pass
            # check if on demand cvmfs provisioning is requested/enabled
            if ondemand_cvmfs != 0:
```

File: creation/lib/cgWParamDict.py [hunk 4]
```diff
@@ -369,34 +389,29 @@ def populate(self, other=None):
...
+                    # check the dir containing cvmfsexec distros to see if they were built previously
+                    if os.path.exists(os.path.join(self.work_dir, "cvmfsexec/tarballs")) and os.listdir(
+                        os.path.join(self.work_dir, "cvmfsexec/tarballs")
+                    ):
+                        # cvmfsexec distros were found from a previous factory reconfig
+                        print(f"...Found cvmfsexec distributions in {os.path.join(self.work_dir)}")
+                        print("......RECOMMENDED: Rebuild distributions using the latest version of cvmfsexec")
+                    else:
+                        print("...cvmfsexec distributions unavailable but on-demand CVMFS requested; Aborting!")
```
Comment (mambelli) [inline creation/lib/cgWParamDict.py:412]: Add "via GLIDEIN_USE_CVMFSEXEC"

File: creation/lib/cgWParamDict.py (hunk 2 above)
Comment (namrathaurs) [reply creation/lib/cgWParamDict.py:394]: Updated

File: creation/lib/cgWParamDict.py (hunk 4 above)
Comment (namrathaurs) [reply creation/lib/cgWParamDict.py:412]: Updated as suggested

File: creation/lib/cgWParamDict.py (hunk 3 above)
Comment (namrathaurs) [reply creation/lib/cgWParamDict.py:403]: Modified as suggested.

File: creation/lib/cgWParamDict.py (hunk 1 above)
Comment (namrathaurs) [reply creation/lib/cgWParamDict.py]: Updated as suggested above

**Comment by mambelli** [comment]
Code con Namratha. Merging without review

**Pull request merged**